### DISCUSSION TITLE
[QMP] Duda sobre formato de entregas

### DISCUSSION THREAD

**Discussion by Darkonis13** [dds-jv/foro#234] [Consultas generales] [answered]
Buenas, quería consultar si los diagramas siguen siendo una forma valida para responder estos enunciados.
Vengo entregándolos así desde la primer iteración, pero ahora se implementa una API... ¿Se puede expresar la API en el diagrama, o implementarla es parte del ejercicio?

**Comment by flbulgarelli** [comment, marked as answer]
¡Hola!

Los diagramas de clases siempre te ayudan a vos a plantear las ideas y a quien lo corrige para entenderlas. Más allá de eso: 

1. No, no es obligatorio un formato particular de entrega para QMP, puede ser código, pseudócódigo, diagramas y explicaciones aunque lo más provechoso será una combinación de diagramas y (pseudo)código. 
2. No hay que implementar el API en el sentido de hacer que funcione la integración son el sistema externo. A nivel diagrama y código por igual deberías declarar una `interface` de Java, y hacer que tus componentes dialoguen con la misma. 

¡Saludos!

**Reply by Darkonis13** [reply]
Muchas gracias profe!
